### DISCUSSION TITLE
How to trace spring-cloud-gateway 2.0.4-release when using skywalking-agent 8.16.0

### DISCUSSION THREAD

**Discussion by forgivelyj** [apache/skywalking#11628] [Q&A]
I tried to use skywalking to track the entire link of the microservice platform. After using skywalking-agent, the call chain between my services has been tracked, including feign calls. But when viewing through the gateway, only a single span can be seen. I have added apm-spring-cloud-gateway-2.0.x-plugin-8.16.0 and apm-spring-webflux-5.x-plugin-8.16.0 to the skywalking-agent plugins. During the debugging process, I found that The traceId of the gateway can be seen in NettyRoutingFilterInterceptor, but the traceId is lost in the HttpClientRequestInterceptor after forwarding.

![1701775196012](https://github.com/apache/skywalking/assets/38929446/bdecc539-d1ef-4a74-87e3-ec9575a05881)

Thanks for the help afterwards

**Comment by wu-sheng** [comment]
I am not sure which version are you using. We only tested 2.0.4 through here, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/gateway-2.0.x-scenario You could cross-check the use. Usually, if you are using a custom filter, this is not an out-of-box thing, you need some changes in your codes.

There are more cases provided in the folder of test to help you cross check the differences

https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios

<img width="282" alt="image" src="https://github.com/apache/skywalking/assets/5441976/54199e36-c8c2-4272-aba4-3ffbf951e960">

Tips, Gateway is complex, so, check carefully and debug more. You know more than us about how the codes are running and forward your request.

**Reply by forgivelyj** [reply]
I tried to delete all the custom code in the gateway, but I still can’t see the span of the gateway and the subsequent links of the service. I think I need further debugging. Thank you again for your suggestions.

![1701837070354](https://github.com/apache/skywalking/assets/38929446/76766f59-7f30-4856-87db-8a7cad769f4d)

**Reply by wu-sheng** [reply]
The demo codes are there. Check the links I shared.

**Reply by forgivelyj** [reply]
I'm sorry for misunderstanding. What I meant was that the demo you provided was normal. I tried to delete my custom code without success, so I still need to further debug my code. Thank you for your suggestion
